src/Workspaces/Core/Portable/FindSymbols/FindReferences/DependentTypeFinder.cs

```diff
@@ -50,7 +50,8 @@ internal static partial class DependentTypeFinder
 
         private static readonly RelatedTypeCache s_typeToImmediatelyDerivedClassesMap = new RelatedTypeCache();
         private static readonly RelatedTypeCache s_typeToTransitivelyDerivedClassesMap = new RelatedTypeCache();
-        private static readonly RelatedTypeCache s_typeToTransitivelyImplementingTypesMap = new RelatedTypeCache();
+        private static readonly RelatedTypeCache s_typeToTransitivelyImplementingStructuresAndClassesMap = new RelatedTypeCache();
```

Comment:
s_typeToTransitivelyImplementingStructuresClassesAndInterfacesMap seems just superset of s_typeToTransitivelyImplementingStructuresAndClassesMap 

so, it would do same search twice and cache same data twice.

can we add filter instead here?
http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/FindSymbols/FindReferences/DependentTypeFinder.cs,117
by giving the method (FindTypesFromCacheOrComputeAsync - http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/FindSymbols/FindReferences/DependentTypeFinder.cs,56) some kind of predicate/filter for symbol?